### PR TITLE
Review parsing of peer URI

### PR DESCRIPTION
**Pull request by lucafavatella**
Finishes https://www.pivotaltracker.com/story/show/155700294
Fits in the release notes line "Improves the stability of the testnet".

### CODE REVIEW

File: apps/aecore/src/aec_peers.erl [hunk 1]
```diff
@@ -685,6 +685,15 @@ parse_uri(Uri) ->
             Error
     end.
 
+uri_safe_parse(Uri) when is_binary(Uri) -> %% Workaround for https://github.com/erlang/otp/pull/1739
```
Comment (lucafavatella) [inline apps/aecore/src/aec_peers.erl]: Planned removal tracked as https://www.pivotaltracker.com/story/show/155774525

File: apps/aecore/src/aec_peers.erl [hunk 2]
```diff
@@ -101,7 +99,7 @@ valid_uris(Uris) ->
 %%------------------------------------------------------------------------------
 %% Add peer by url. Connect if `Connect==true`
 %%------------------------------------------------------------------------------
--spec add(http_uri_uri() | peer(), boolean()) -> ok | {error, any()}.
```
Comment (ThomasArts) [inline apps/aecore/src/aec_peers.erl]: well spotted!

File: apps/aecore/src/aec_peers.erl [hunk 3]
```diff
@@ -685,6 +683,15 @@ parse_uri(Uri) ->
             Error
     end.
 
+uri_safe_parse(Uri) when is_binary(Uri) -> %% Workaround for https://github.com/erlang/otp/pull/1739
+    uri_safe_parse(unicode:characters_to_list(Uri, utf8));
+uri_safe_parse(Uri) ->
+    F = aeu_uri:scheme_validation_fun(valid_uri_schemes()),
+    http_uri:parse(Uri, [{scheme_validation_fun, F}]).
+
+valid_uri_schemes() ->
```
Comment (ThomasArts) [inline apps/aecore/src/aec_peers.erl]: would it make sense to also add HTTP capitalized?

Comment (lucafavatella) [reply apps/aecore/src/aec_peers.erl]: I pushed a commit meant to address this, CI is running.

**Comment by lucafavatella** [comment]
> While you're on this, is this implementation supports HTTP Basic Auth? e.g. user:pass@127.0.0.1:3013 ? Or at least id@127.0.0.1:3013 ? I think we'll need that very soon.

@dincho I see that way to pass user info is deprecated. Though it works, [without splitting on `:`](https://github.com/erlang/otp/blob/OTP-20.2.4/lib/inets/src/http_lib/http_uri.erl#L207).

Positive cases are [tested upstream](https://github.com/erlang/otp/blob/OTP-20.2.4/lib/inets/test/uri_SUITE.erl#L121-L126) in current http_uri:parse . Also I ran some samples:
```
6> http_uri:parse("user:pass@127.0.0.1:3013", [{scheme_validation_fun, fun("http") -> valid; (S) -> {error, {bad_scheme, S}} end}]).
{error,{bad_scheme,"user"}}
7> http_uri:parse("http://user:pass@127.0.0.1:3013", [{scheme_validation_fun, fun("http") -> valid; (S) -> {error, {bad_scheme, S}} end}]).
{ok,{http,"user:pass","127.0.0.1",3013,"/",[]}}
8> http_uri:parse("http://id@127.0.0.1:3013", [{scheme_validation_fun, fun("http") -> valid; (S) -> {error, {bad_scheme, S}} end}]).
{ok,{http,"id","127.0.0.1",3013,"/",[]}}
9> http_uri:parse(<<"http://user:pass@127.0.0.1:3013">>).
{ok,{http,<<"user:pass">>,<<"127.0.0.1">>,3013,<<"/">>,
          <<>>}}
10> http_uri:parse(<<"http://id@127.0.0.1:3013">>).
{ok,{http,<<"id">>,<<"127.0.0.1">>,3013,<<"/">>,<<>>}}
11> http_uri:parse(<<"http://user:pass:a@127.0.0.1:3013">>).
{ok,{http,<<"user:pass:a">>,<<"127.0.0.1">>,3013,<<"/">>,
          <<>>}}
12> http_uri:parse(<<"http://user:pass:a:b@127.0.0.1:3013">>).
{ok,{http,<<"user:pass:a:b">>,<<"127.0.0.1">>,3013,<<"/">>,
          <<>>}}
```

**Pull request merged**